### PR TITLE
S3 optional metadata upload parameter

### PR DESCRIPTION
**Pull request by rossbeale**
For #359 - metadata now optional.

### CODE REVIEW

File: lib/fastlane/actions/s3.rb [hunk 1]
```diff
@@ -271,6 +276,11 @@ def self.available_options
                                        description: "zipped .dsym package for the build ",
                                        optional: true,
                                        default_value: Actions.lane_context[SharedValues::DSYM_OUTPUT_PATH]),
+          FastlaneCore::ConfigItem.new(key: :upload_metadata,
+                                       env_name: "",
+                                       description: "upload relevant metadata for this build ",
```
Comment (KrauseFx) [inline lib/fastlane/actions/s3.rb]: Please remove the space at the end and change the first word to upper case.

**Comment by KrauseFx** [comment]
Looks good to me, I'd just like @joshdholtz' confirmation as he did the original implementation. :+1: Thanks for your contribution @rossbeale.

File: docs/Actions.md [hunk 2]
```diff
@@ -408,7 +408,8 @@ s3(
   bucket: ENV['S3_BUCKET'],                       # Required from user.
   file: 'AppName.ipa',                            # This would come from IpaAction.
   dsym: 'AppName.app.dSYM.zip',                   # This would come from IpaAction.
-  path: 'v{CFBundleShortVersionString}_b{CFBundleVersion}/', # This is actually the default.
+  path: 'v{CFBundleShortVersionString}_b{CFBundleVersion}/', # This is actually the default.,
```
Comment (joshdholtz) [inline docs/Actions.md]: Looks like there is an extra comment at the end of this comment?

Comment (KrauseFx) [inline docs/Actions.md]: * comma?

Comment (rossbeale) [inline docs/Actions.md]: Resolved. :)

File: lib/fastlane/actions/s3.rb (hunk 1 above)
Comment (rossbeale) [inline lib/fastlane/actions/s3.rb]: Done :)

**Comment by joshdholtz** [comment]
@rossbeale Looks good to me :+1: Thanks for adding this :dolphin:

**Comment by KrauseFx** [comment]
Wow, what a team work. Thanks everybody, this will be in the next release :+1:

**Pull request merged**